### PR TITLE
enable hwmon for sensor collection for bare metal clusters

### PR DESCRIPTION
**Pull request by dhellmann**
Enable the hwmon data collection so that hardware telemetry like CPU
temperature and fan speeds are available for bare metal clusters.

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: assets/node-exporter/daemonset.yaml [hunk 1]
```diff
@@ -23,7 +23,6 @@ spec:
         - --path.sysfs=/host/sys
         - --path.rootfs=/host/root
         - --no-collector.wifi
-        - --no-collector.hwmon
```
Comment (lilic) [inline assets/node-exporter/daemonset.yaml:26]: This modifies the generate assets directly, you need to modify the jsonnet file to remove this, see this guide how to then generate https://github.com/openshift/cluster-monitoring-operator/blob/master/CONTRIBUTING.md

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
   - Adjusted NodeClockNotSynchronising, NodeNetworkReceiveErrs, and NodeNetworkTransmitErrs alerts.
 - [#962](https://github.com/openshift/cluster-monitoring-operator/pull/962) Enable namespace by pod and pod total networking Grafana dashboards.
 - [#959](https://github.com/openshift/cluster-monitoring-operator/pull/959) Remove memory limits from prometheus-config-reloader in user workload monitoring
+- [#971](https://github.com/openshift/cluster-monitoring-operator/pull/971) Enable `hwmon` in node-exporter for hardware sensor data collection
```
Comment (lilic) [inline CHANGELOG.md:10]: I remember us disabling not urgent collectors due to high cardinality, I am wondering how many new series this brings in on each cluster?

Comment (dhellmann) [reply CHANGELOG.md:10]: The number of series depends on the number of sensors visible on the node, so it's hard to pin that down. On my dev system, I only get some temperature sensors. On other hosts, we should see more.

Comment (s-urbaniak) [reply CHANGELOG.md:10]: @dhellmann I am wondering is this enablement part of some concrete initiative?

File: assets/node-exporter/daemonset.yaml (hunk 1 above)
Comment (dhellmann) [reply assets/node-exporter/daemonset.yaml:26]: Hmm, this was the only file in this repository that had that string in it. Where is this file coming from upstream?

File: CHANGELOG.md (hunk 2 above)
Comment (dhellmann) [reply CHANGELOG.md:10]: Here's some example output from one host: https://paste.centos.org/view/e2d24856

That includes temperature data, but that host is apparently not exposing fan speed or other sensors.

File: assets/node-exporter/daemonset.yaml (hunk 1 above)
Comment (paulfantom) [reply assets/node-exporter/daemonset.yaml:26]: PR that removed `hwmon` from enabled collectors in upstream repository: https://github.com/prometheus-operator/kube-prometheus/pull/381

**Comment by dhellmann** [comment]
I've updated the PR with a jsonnet expression to remove the flag from the upstream version of the deployment settings. 

Regenerating the assets modified manifests/0000_50_cluster-monitoring-operator_02-role.yaml but it looks like that has to do with something not sorting output consistently. I can manually remove that change, but thought I should include all of the output of `make generate` to start out.

**Comment by s-urbaniak** [comment]
@dhellmann this lgtm, but could you please rebase?

**Comment by dhellmann** [comment]
Rebased to resolve the changelog merge conflict.

**Comment by s-urbaniak** [comment]
just holding to verify that @lilic's comments have been addressed.

**Comment by lilic** [comment]
/hold cancel

lgtm 👍

**Comment by dhellmann** [comment]
@lilic The test failures look real, but I don't know how they're related to this change. I see similar failures on https://github.com/openshift/cluster-monitoring-operator/pull/965. Is it possible something else broke that test job?

**Pull request merged**